SalesforceCommerceCloud/pwa-kit
# @W-16795135@ [App Extensibility ⚙️] Use ApplicationExtension abstract class over Typescript interface

# Description

In this PR we are replace the use of TypeScript interface "IApplicationExtension" with an abstract class "ApplicationExtension". This allows use to hid some of the implementation for methods like `getName` and `getConfig` and also leaves the door open to adding more features in the future that all extensions will consume when upgrading their dependencies.

# Changes

- Replace IApplicationExtension with ApplicationExtension abstract class
- Update tests
- 

# How to Test-Drive This PR

- If tests are passing this is working.
- But if you want, run the dev server and check that the express endpoint `/sample` still works.

# Checklists

## General
